plugins/lighthouse/src/api.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiRef } from '@backstage/core';
+
+export type LighthouseCategoryId =
+  | 'pwa'
+  | 'seo'
+  | 'performance'
+  | 'accessibility'
+  | 'best-practices';
+
+export interface LighthouseCategoryAbbr {
+  id: LighthouseCategoryId;
+  score: number;
+  title: string;
+}
+
+export interface LASListRequest {
+  offset?: number;
+  limit?: number;
+}
+
+export interface LASListResponse<Item> {
+  items: Item[];
+  total: number;
+  offset: number;
+  limit: number;
+}
+
+interface AuditBase {
+  id: string;
+  url: string;
+  timeCreated: string;
+}
+
+export interface AuditRunning extends AuditBase {
+  status: 'RUNNING';
+}
+
+export interface AuditFailed extends AuditBase {
+  status: 'FAILED';
+  timeCompleted: string;
+}
+
+export interface AuditCompleted extends AuditBase {
+  status: 'COMPLETED';
+  timeCompleted: string;
+  report: Object;
+  categories: Record<LighthouseCategoryId, LighthouseCategoryAbbr>;
+}
+
+export type Audit = AuditRunning | AuditFailed | AuditCompleted;
+
+export interface Website {
+  url: string;
+  audits: Audit[];
+  lastAudit: Audit;
+}
+
+export type WebsiteListResponse = LASListResponse<Website>;
+
+export interface TriggerAuditPayload {
+  url: string;
+  options: {
+    lighthouseConfig: {
+      settings: {
+        emulatedFormFactor: string;
+      };
+    };
+  };
+}
+
+export class FetchError extends Error {
```

Comment:
I think it's nice to set the `name` prop of the error as well, something like https://github.com/spotify/backstage/blob/bc4f6ea3ad05df49e4278b0b0da41207bb533eca/packages/cli/src/helpers/errors.ts#L20